cachi2/core/resolver.py

```diff
@@ -22,7 +28,36 @@
 supported_package_managers = list(_package_managers)
 
 
+@contextmanager
+def make_source_dir_work_copy(request: Request) -> Iterator:
+    """Switches the source_dir of a Request for a full copy."""
+    original_source_dir = request.source_dir
+    copy_dir_name = "./.cachi2-source-copy-" + secrets.token_hex(8)
```

Comment:
Why do we want to handle creation of a uniquely-named directory on our own when there's `tempfile.TemporaryDirectory`?

Reply:
E.g. `tempfile.mkdtemp(prefix="cachi2-source-copy-", dir=./)` (the `dir` arg is optional and defaults to system-specifc temp dir(s), but it can be anything)

Reply:
> E.g. tempfile.mkdtemp(prefix="cachi2-source-copy-", dir=./) (the dir arg is optional and defaults to system-specifc temp dir(s), but it can be anything)

one doesn't even need the low-level C-wrapping API, even the high-level Python API is totally fine for our case :)

Reply:
That *is* [the Python API ](https://docs.python.org/3.9/library/tempfile.html#tempfile.mkdtemp)... or do you mean [`tempfile.TemporaryDirectory`](https://docs.python.org/3.9/library/tempfile.html#tempfile.TemporaryDirectory)?

Reply:
My bad, I was under the false impression that `tempfile.mkdtemp` specifically is just one of the Python API bindings over some C implementation counterpart, in this case https://linux.die.net/man/3/mkdtemp or the corresponding series of syscalls. Turns out, I was wrong and [tempile](https://github.com/python/cpython/blob/main/Lib/tempfile.py#L344) is pure Python code after all, sorry for the noise.